### PR TITLE
Manual backport for #11108 (Don't re-verify invalid FITS cards after their values have been replaced)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to manually backport #11108 to v4.2.x

### CODE REVIEW

**Comment by pllim** [comment]
I am not 100% sure about this, please review carefully.

File: CHANGES.rst [hunk 1]
```diff
@@ -1105,6 +1105,9 @@ astropy.io.ascii
 astropy.io.fits
 ^^^^^^^^^^^^^^^
 
+- Fix bug where manual fixes to invalid header cards were not preserved when
```
Comment (pllim) [inline CHANGES.rst:1108]: This is still listed under 4.0.5. Do I have to move it as part of the manual backport?

Comment (eteq) [reply CHANGES.rst:1108]: No, it should stay here just as you've done.

**Review by eteq — COMMENTED** [review]
This looks right to me assuming the tests pass... *but* I admit I'm not following all the logic, so @saimn or @embray might want to weigh in...

**Review by saimn — APPROVED** [review]
The io.fits failure was fixed by #11438 so I think it's all good for this one, the remaining failures are unrelated.

**Pull request merged**